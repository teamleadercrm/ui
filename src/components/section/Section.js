import React from 'react';
import PropTypes from 'prop-types';
import Box from '../box';
import cx from 'classnames';
import theme from './theme.css';

const SIZES = {
  small: 3,
  medium: 4,
  large: 5,
};

/** @type {React.ComponentType<any>} */
const Section = ({ children, className, color, size, ...rest }) => {
  const classNames = cx(theme['section'], className, theme[color]);

  return (
    <Box data-teamleader-ui="section" className={classNames} element="section" padding={SIZES[size]} {...rest}>
      {children}
    </Box>
  );
};

Section.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  color: PropTypes.oneOf(['white', 'neutral', 'mint', 'violet', 'ruby', 'gold', 'aqua']),
  size: PropTypes.oneOf(Object.keys(SIZES)),
};

Section.defaultProps = {
  color: 'white',
  size: 'medium',
};

export default Section;
