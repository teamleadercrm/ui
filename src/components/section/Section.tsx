import cx from 'classnames';
import React, { ReactNode } from 'react';
import { GenericComponent } from '../../@types/types';
import Box from '../box';
import { BoxProps, Padding } from '../box/Box';
import theme from './theme.css';

type Color = 'white' | 'neutral' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua';
type Size = 'small' | 'medium' | 'large';

const PADDINGS: Record<Size, Padding> = {
  small: 3,
  medium: 4,
  large: 5,
};

interface SectionProps extends Omit<BoxProps, 'ref'> {
  children?: ReactNode;
  color?: Color;
  size?: Size;
}

const Section: GenericComponent<SectionProps> = ({
  children,
  className,
  color = 'white',
  size = 'medium',
  ...rest
}) => {
  const classNames = cx(theme['section'], className as string, theme[color]);

  return (
    <Box data-teamleader-ui="section" className={classNames} element="section" padding={PADDINGS[size]} {...rest}>
      {children}
    </Box>
  );
};

export default Section;
