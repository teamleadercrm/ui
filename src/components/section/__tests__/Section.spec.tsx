import React from 'react';
import { render } from '@testing-library/react';
import Section from '../Section';

describe('Component - Section', () => {
  it('renders', async () => {
    const { asFragment } = render(<Section>This is a Section</Section>);
    expect(asFragment()).toMatchSnapshot();
  });
});
