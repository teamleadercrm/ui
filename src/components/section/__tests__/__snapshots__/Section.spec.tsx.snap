// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Section renders 1`] = `
<DocumentFragment>
  <section
    class="box padding-bottom-4 padding-left-4 padding-right-4 padding-top-4 section white"
    data-teamleader-ui="section"
  >
    This is a Section
  </section>
</DocumentFragment>
`;
